CMakeLists.txt

```
@@ -1,12 +1,20 @@
 cmake_minimum_required(VERSION 3.16)
 project(deltachat LANGUAGES C)
 
+if(APPLE)
+	set(DYNAMIC_EXT "dylib")
+elseif(UNIX AND NOT APPLE)
```

```suggestion
elseif(UNIX)
```
not needed due to `elseif`

---

Tried that, would not work

---

Why? Work on Linux for me, sets `DYNAMIC_EXT` to `so`.

---

It sets it on Darwin to so as well (without the not apple)

---

Sure on Darwin `UNIX` is true, but there is an `if(APPLE)` right above, so if `APPLE` is true, it goes into the first branch. So in this check `APPLE` is always `False`, therefore `UNIX AND NOT APPLE = UNIX AND True = UNIX`.

---

That's what I thought as well. As I tried at that way, CMake set it to so (I can test it again)

---

Yes, please test again on darwin. I don't see why this could be needed unless `if` in CMake doesn't work like in pretty much any other language (C, Python, Pascal, R, D, Rust, ...).

---

Maybe @Simon-Laux can help?

---

Oh sorry, forgot this. Will try after work today

---

@Kloenk any updates?

---

@Kloenk ping :)

---

@Kloenk ping :)